Updating changelog for 1.1.0 release (#1081)
Azure/azure-sdk-for-android: sdk/communication/azure-communication-chat/CHANGELOG.md

### Other Changes

## 1.1.0 (2022-04-11)
### Features Added
- Added support for metadata in messages.
- Added options class `TypingNotificationOptions` for setting `SenderDisplayName` of the notification sender.
- Added `SenderDisplayName` to `TypingIndicatorReceivedEvent`.
- Added support FCM push notifications:

>     - Added methods to `ChatAsyncClient`:
>         - `startPushNotifications`
>         - `stopPushNotifications`
>         - `handlePushNotification`
>         - `addPushNotificationHandler`
>         - `removePushNotificationHandler`
>     - Added methods to `ChatClient`:
>         - `startPushNotifications`
>         - `stopPushNotifications`
>         - `handlePushNotification`
>         - `addPushNotificationHandler`
>         - `removePushNotificationHandler`

Reviewer: We might want to collapse both lists into one as they're almost identical.
```suggestion
    - Added the following methods to `ChatAsyncClient` and `ChatClient`:
        - `startPushNotifications`
        - `stopPushNotifications`
        - `handlePushNotification`
        - `addPushNotificationHandler`
        - `removePushNotificationHandler`
```

     - Added `ChatPushNotification` as the type for passing a push notification payload.

>      - Added instructions in README.md in sample app for steps to setup FCM push notifications.

Reviewer: ```suggestion
     - Added instructions in the README.md of our sample app on how to setup FCM push notifications.
```

     - Added code in sample app for FCM push notification end to end scenario.

> - Added `startRealtimeNotifications(Context context, Consumer<Throwable> errorHandler)` to `ChatAsyncClient`.
> - Added `startRealtimeNotifications(Context context, Consumer<Throwable> errorHandler)` to `ChatClient`.

Reviewer: ```suggestion
- Added `startRealtimeNotifications(Context context, Consumer<Throwable> errorHandler)` to `ChatAsyncClient` and `ChatClient`.
```

### Bugs Fixed

> - Fix `azure-communication-common` dependency issue.
> - Fix a BUG in real-time notification payload for GCCH

Reviewer: We might also want to add details on what these issues were about. Linking to the PR that fixed them is also nice to have. None of this is required though.
```suggestion
- Fixed a dependency issue with `azure-communication-common`.
- Fixed a BUG in real-time notification payload for GCCH.
```

### Other Changes

> - Deprecate `startRealtimeNotifications(String skypeUserToken, Context context)` in `ChatAsyncClient`.
> - Deprecate `startRealtimeNotifications(String skypeUserToken, Context context)` in `ChatClient`.

Reviewer: I'd suggest collapsing both points into one and pointing users to the method that is replacing the one being deprecated.
```suggestion
- Deprecated `startRealtimeNotifications(String skypeUserToken, Context context)` in `ChatAsyncClient` and `ChatClient`. Users should utilize `startRealtimeNotifications(Context context, Consumer<Throwable> errorHandler)` instead.
```
